src/module_wrap.cc

```diff
@@ -47,12 +45,17 @@ ModuleWrap::ModuleWrap(Environment* env,
 }
 
 ModuleWrap::~ModuleWrap() {
-  Local<Module> module = module_.Get(Isolate::GetCurrent());
-  std::vector<ModuleWrap*>* same_hash = module_map_[module->GetIdentityHash()];
-  auto it = std::find(same_hash->begin(), same_hash->end(), this);
+  Isolate* isolate = Isolate::GetCurrent();
+  Environment* env = Environment::GetCurrent(isolate);
+  Local<Module> module = module_.Get(isolate);
+  std::vector<ModuleWrap*>& same_hash =
+      env->module_map[module->GetIdentityHash()];
+  auto it = std::find(same_hash.begin(), same_hash.end(), this);
 
-  if (it != same_hash->end()) {
-    same_hash->erase(it);
+  if (it != same_hash.end()) {
+    same_hash.erase(it);
+    if (same_hash.empty())
+      env->module_map.erase(module->GetIdentityHash());
   }
```

Comment:
Correct me if I'm wrong but if you used a `std::unordered_multimap`, you could drop the extra indirection of the `std::vector`.  You could then simply look up `env->module_map.equal_range(module->GetIdentityHash())`.

Reply:
Yes, good idea – done.